timemachine/cpp/CMakeLists.txt

```diff
@@ -9,6 +9,8 @@ project(timemachine LANGUAGES CXX CUDA)
 find_package(PythonInterp 3.7 REQUIRED)
 find_package(PythonLibs 3.7 REQUIRED)
 
+set(CMAKE_CXX_FLAGS "-Werror -Wall -Wextra")
```

Comment:
Should we append to `CMAKE_CXX_FLAGS` instead of setting? (similar to the handling of `CMAKE_CUDA_FLAGS` below)